components/eamxx/src/dynamics/homme/eamxx_homme_iop.cpp

```diff
@@ -428,17 +482,144 @@ apply_iop_forcing(const Real dt)
         // For BFB on restarts, Q needs to be updated after we compute Qdp
         Q_i(q, ilev) = Qdp_i(q, ilev)/dp3d_i(ilev);
       });
+      team.team_barrier();
 
-      // Convert updated temperature back to potential temperature
+      // Recompute rstar with updated qv, and convert updated temperature back to potential temperature
       elem_ops.get_R_star(kv, use_moisture, qv_scalar, rstar_scalar);
       Kokkos::parallel_for(Kokkos::ThreadVectorRange(team, NLEV), [&] (const int k) {
         vtheta_dp_i(k) = temperature_i(k)*rstar_i(k)*dp3d_i(k)/(Rair*exner_i(k));
       });
     });
-
-    // Release WS views
-    ws.release_macro_block(rstar_slot, NGP*NGP);
   });
+
+  if (iop_nudge_tq or iop_nudge_uv) {
```

Comment:
Below is port of `iop_apply_relaxation()`. Like the F90, I can put it in it's own function if that makes it easier to read.